### ISSUE TITLE
ansible.builtin.import_tasks -> syntax-check[missing-file]: Unable to retrieve file contents

### ISSUE THREAD

**Issue by tachtler** [ansible/ansible-lint#3459]
##### Summary

If I use ansible.builtin.import_tasks and the directory structure is defined under `~/git/ansible/playbooks/tasks` a YAML file with the name e.g. `gather_facts.yml `can be included correctly using ansible as follows.

```
---
- name: Task tasks/gather_facts.yml
  ansible.builtin.import_tasks: tasks/gather_facts.yml
```

However, ansible-lint returns the following error:

```
syntax-check[missing-file]: Unable to retrieve file contents
playbooks/roles/role_test:1:1 Could not find or access '/tmp/tasks/gather_facts.yml' on the Ansible Controller. 
```

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 6.15.1.dev45 using ansible 2.15.0
```

- ansible installation method: Archlinux pacman
- ansible-lint installation method: Archlinux pacman

##### STEPS TO REPRODUCE

```console (paste below)
ansible-lint
syntax-check[missing-file]: Unable to retrieve file contents
playbooks/roles/role_test:1:1 Could not find or access '/tmp/tasks/gather_facts.yml' on the Ansible Controller. 
```

##### Desired Behavior

I will expect, that **NOT** `/tmp` will be assumed!

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behavior

Please give some details of what is happening.
Include a [minimum complete verifiable example] with:

- minimized playbook to reproduce the error
- the output of running ansible-lint including the command line used
- if you're getting a stack trace, also the output of
  `ansible-playbook --syntax-check playbook`

```paste below
$ ansible-playbook -i ~/ansible/playbooks/inventories/production ~/ansible/playbooks/play.yml --syntax-check

playbook: /home/klaus/ansible/playbooks/play.yml
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by Hennns** [comment]
Experiencing the same issue, adding some more information:

It works on ansible-lint 6.14 
> ansible-lint --version
ansible-lint 6.14.4 using ansible 2.14.4

However, it stopped working on ansible-lint 6.15
> ansible-lint --version
ansible-lint 6.15.0 using ansible 2.14.5

When running with -v I can see that the directory is correct:
> /code# ansible-lint -v
INFO     Identified /code as project root due config file /code/.config/ansible-lint.yml.

If it would be helpful, I think it should be pretty easy to create a mini project, which I would be able to share, that will reproduce this error.

**Comment by Hennns** [comment]
I have added an example here: https://github.com/Hennns/Example

**Comment by ajinkyau** [comment]
@Hennns Using your example, If I run `ansible-lint` on the base dir `Examples` or it's child dirs like `code` or `production` I get an error.
But when I run `ansible-lint roles -vv` (i.e: specifically on the roles dir) it executes without any errors.
I think this is probably because of creating a temporary playbook if it's a role:
https://github.com/ansible/ansible-lint/blob/main/src/ansiblelint/rules/syntax_check.py#L89
which is why it looks for imports in the `tmp` dir.

**Comment by cpascual** [comment]
@Hennns  wrote:
> It works on ansible-lint 6.14

This saved my day: I was hit by this bug via the [ansible-lint-action](https://github.com/ansible/ansible-lint-action), and the workaround was to pint it to `ansible/ansible-lint-action@v6.14.4`

(I hope this helps other people)
